packages/react/src/lib/factories.ts

```diff
@@ -164,6 +178,13 @@ function createShorthandFromValue(
     props[mappedProps[overrideProps.as || defaultProps.as] || mappedProp || 'children'] = value
   }
 
+  // Map prop for array value
+  if (valIsArrayObject) {
+    props[
+      mappedProps[overrideProps.as || defaultProps.as] || mappedPropAsArray || 'children'
```

Comment:
We have partially the same condition on 178 line. Can we extract `mappedProps[overrideProps.as || defaultProps.as]` to something meaningful?